### ISSUE TITLE
test_github_25 fails on several CPU architectures

### ISSUE THREAD

**Issue by olebole** [brandon-rhodes/pyephem#99]
Hi,

I created a [Debian package for pyephem](https://tracker.debian.org/pkg/pyephem). Unfortunately, on several (non-intel) architectures, the following test fails:

```
FAIL: test_github_25 (ephem.tests.test_github_issues.GitHubIssues)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "ephem/tests/test_github_issues.py", line 58, in test_github_25
    self.assertRaises(RuntimeError, lambda: sat.neverup)
AssertionError: RuntimeError not raised
```

Full build log [for PowerPC here](https://buildd.debian.org/status/fetch.php?pkg=pyephem&arch=powerpc&ver=3.7.6.0-1&stamp=1454679502).
This happens on  [PowerPC](https://buildd.debian.org/status/fetch.php?pkg=pyephem&arch=powerpc&ver=3.7.6.0-1&stamp=1454679502), [arm (little endian)](https://buildd.debian.org/status/fetch.php?pkg=pyephem&arch=armel&ver=3.7.6.0-1&stamp=1454699774), [arm (little endian, hardware FP unit)](https://buildd.debian.org/status/fetch.php?pkg=pyephem&arch=armhf&ver=3.7.6.0-1&stamp=1454706830), [mips (little endian)](https://buildd.debian.org/status/fetch.php?pkg=pyephem&arch=mipsel&ver=3.7.6.0-1&stamp=1454715633), and [hppa](https://buildd.debian.org/status/fetch.php?pkg=pyephem&arch=hppa&ver=3.7.6.0-1&stamp=1454685021). Since there are little endian and big endian architectures among it, it (contrary to what I suggested before) seems to be not endian dependent.

**Comment by brandon-rhodes** [comment]
If I recall, that was a very dodgy issue involving the exact way that a mathematical loop did or did not spiral in towards a solution for a very particular set of satellite circumstances — the satellite routines in `libastro`, that sits beneath PyEphem, are old and a bit brittle.

Let me know if Debian has any facilities on which I could try testing on these platforms. I have no access to them myself.

**Comment by olebole** [comment]
While in principle this could be solved like for #100, I am wondering about the first lines of the test:
```
        if sys.maxsize > 2147483647:  # breaks under 64 bits, as on Travis-CI
            return
```
which looks like a pragmatic workaround for the amd64 platform. Could it be that the runtime error anyway does not always happen?
As a workaround, I replaced the `self.assertRaises(RuntimeError, lambda: sat.neverup)`  line with
```
        try:
            sat.neverup
        except:
            pass
```
so that it just checks whether it hangs there, but is ignorant of when it raises an exception. I am however wondering why this behaves (known) differently on different platforms?

**Comment by brandon-rhodes** [comment]
Returning to this five years later, it looks like the `sys.maxsize` test no longer exists. Hoping that the situation is therefore resolved, I'm going to go ahead and close this; please re-open if the issue is still somehow a problem!
